src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -859,14 +859,18 @@ public bool TryGetData(string column, out object value)
             }
         }
 
-        internal async Task<MigrationAssessmentResult> GetAssessmentItems(string[] connectionStrings, string xEventsFilesFolderPath)
+        internal async Task<MigrationAssessmentResult> GetAssessmentItems(string[] connectionStrings, string xEventsFilesFolderPath, bool collectAdhocQueries)
         {
             SqlAssessmentConfiguration.EnableLocalLogging = true;
             SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath = Path.GetDirectoryName(Logger.LogFileFullPath);
 
             SqlConnectionLocator locator = new SqlConnectionLocator();
+            AdHocQueriesCollectionParameters parameters = new()
```

Comment:
Looks good, hope you have verified these changes as here we are introducing a new object.

Reply:
### yes